docs/proposals/csi-deployment-crd.md

```diff
@@ -0,0 +1,258 @@
+# Proposal:  CSI deployment CRD
+
+Last updated: 05.03.20
+
+## Abstract
+
+To deploy CSI component with operator, it's necessary to provide CRD so operator can deploy CSI programmable using its configuration 
+
+## Background
+
+Currently CSI is deployed using helm charts and value
+
+## Proposal
+
+Example of CRD:
+```
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: csi.dell.com
+  labels:
+    app.kubernetes.io/name: csi-operator
+spec:
+  group: csi.dell.com
+  names:
+    kind: CSI
+    listKind: CSIList
+    plural: csi
+    singular: csi
+  scope: Cluster
+  versions:
+    - name: v1beta1
+      served: true
+      storage: true
+```
+
+Example of CR:
+```
+apiVersion: csi.dell.com/v1beta1
+kind: CSI
+metadata:
```

Comment:
We also need to specify node selectors and ability to not deploy scheduler extender, don't manage nodes CRD, etc.